lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -115,9 +115,10 @@ impl LocalShard {
 
         let read_handles: Vec<_> = {
             let segments_guard = segments.read();
-            segments_guard
-                .iter()
-                .map(|(_, segment)| {
+            let segments = segments_guard.non_appendable_then_appendable_segments();
```

Comment:
We must enforce ordering here across `spawn_blocking`.

Reply:
Try to resolve this by making reads on all non-appendable segments first, and then on all appendable. Not 100% sure if that's the best/right solution, but feels like it should work for what we try to achieve here?